src/ZoneServer/Skills/Handlers/Archer/TwinArrows.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Threading.Tasks;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.World.Actors;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Archer
+{
+	/// <summary>
+	/// Handles the Archer skill TwinArrows.
+	/// </summary>
+	[SkillHandler(SkillId.Archer_TwinArrows)]
+	public class TwinArrows : ITargetSkillHandler
+	{
+		/// <summary>
+		/// Handles the skill, do two consecutive hits on the enemy.
+		/// </summary>
+		public void Handle(Skill skill, ICombatEntity caster, ICombatEntity designatedTarget)
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.TurnTowards(designatedTarget);
+
+			if (designatedTarget == null)
+			{
+				return;
+			}
+
+			if (!caster.Position.InRange2D(designatedTarget.Position, skill.Data.MaxRange))
+			{
+				caster.ServerMessage(Localization.Get("Too far away."));
+				return;
+			}
+
+			this.Attack(skill, caster, designatedTarget);
+		}
+
+		/// <summary>
+		/// Executes the actual attack.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="designatedTarget"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ICombatEntity designatedTarget)
+		{
+			var damageDelay = TimeSpan.Zero;
+			var skillHitDelay = skill.Data.DefaultHitDelay;
+
+			// TODO: Add more 50% damage to enemies using cloth armor type
+
+			var skillHitResult = SCR_SkillHit(caster, designatedTarget, skill);
```

Comment:
I see you're generating two skill hits here, but the way the game typically handles double-hits is with one modified skill hit that splits the damage in half and shows two hits. You can take a look at how we currently handle this for other skills in `SCR_CalculateDamage`.

Reply:
gotcha